### PR TITLE
feat: improve DD and CD creation parameters

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/procedures/configureDividendDistribution.ts [hunk 1]
```diff
@@ -153,12 +164,17 @@ export async function prepareConfigureDividendDistribution(
     ...corporateActionArgs,
   });
 
-  const rawPortfolioNumber = originPortfolio && numberToU64(originPortfolio.id, context);
+  const rawPortfolioNumber =
```
Comment (monitz87) [inline src/api/procedures/configureDividendDistribution.ts:167]: Use optionize here

File: src/api/procedures/initiateCorporateAction.ts [hunk 2]
```diff
@@ -130,12 +132,14 @@ export async function prepareInitiateCorporateAction(
...
-    )
-  );
+  const rawWithholdings = taxWithholdings
+    ? taxWithholdings.map(({ identity, percentage }) =>
+        tuple(
+          stringToIdentityId(signerToString(identity), context),
+          percentageToPermill(percentage, context)
+        )
+      )
+    : null;
```
Comment (monitz87) [inline src/api/procedures/initiateCorporateAction.ts:142]: You can use `taxWithholdings && taxWithholdings.map(...);` instead of the ternary here

**Review by monitz87 — CHANGES_REQUESTED** [review]

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**